src/FlagTool.cpp

```diff
@@ -78,28 +78,24 @@ FlagTool::FlagTool(std::shared_ptr<cs::core::InputManager> const& pInputManager,
     }
   });
 
-  pMinimized.onChange().connect(
-      [this](bool val) { mGuiItem->callJavascript("setMinimized", val); });
+  pMinimized.connect([this](bool val) { mGuiItem->callJavascript("setMinimized", val); });
 
   mGuiItem->registerCallback("minimizeMe", "Call this to minimize the flag.",
       std::function([this]() { pMinimized = true; }));
   mGuiItem->callJavascript("setActivePlanet", sCenter, sFrame);
-
-  pText.touch();
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 FlagTool::~FlagTool() {
+  auto pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
+  pSG->GetRoot()->DisconnectChild(mGuiTransform.get());
```

Comment:
```suggestion
  if (pSG) {
    pSG->GetRoot()->DisconnectChild(mGuiTransform.get());
  }
```

Reply:
Can it actually happen that the VistaSceneGraph is `null` here?

Reply:
Hopefully not, but better check for it

Reply:
In a properly started application it cannot be null. If we start checking every pointer we get from Vista ...